### DISCUSSION TITLE
Option for enabling/disabling copilot

### DISCUSSION THREAD

**Discussion by 24namanjain** [community/community#10930] [Copilot Conversations] [answered]
Can there be an option for temporarily disable copilot in VS code as sometime I don't want co-pilot to suggest code to me. If the requested  option can just be activated/deactivated on a click like some extension which are always available in the bottom bar but can be simply enabled or disabled on a click.

**Comment by davecheney** [comment, marked as answer]
<img width="492" alt="Screen Shot 2022-02-03 at 8 10 54 am" src="https://user-images.githubusercontent.com/7171/152237975-ea0238e7-1d24-4f35-a5a2-d9b8d194d18e.png">
Click the copilot icon, choose to disable for the current file type, or globally. 
<img width="505" alt="Screen Shot 2022-02-03 at 8 11 16 am" src="https://user-images.githubusercontent.com/7171/152238021-0c243436-caf1-4797-8a21-36baef0c91da.png">
Click again to re-enable.

**Comment by mvasilkov** [comment]
Was this removed at some point? It used to work, but now it doesn't for me.

**Reply by davecheney** [reply]
@mvasilkov the option has not been removed. i'm sorry you are having difficulties. Can you please tell me a bit more about what you tried, and what happened. Thank you.

**Reply by zhaobenx** [reply]
Same problem，I just click the copilot icon, and it turns brown, but the auto completion still exists.

**Reply by mvasilkov** [reply]
> @mvasilkov the option has not been removed. i'm sorry you are having difficulties. Can you please tell me a bit more about what you tried, and what happened. Thank you.

I click on it, nothing happens.

**Reply by davecheney** [reply]
Which version of the extension are you using? Can you please confirm you are using vscode, not oss code or vs codium.

**Reply by matrs** [reply]
Hi @davecheney, 
I'm using `vscode@1.71.1` with `GitHub.copilot@1.44.6735`  in manjaro (arch) linux and I have the same problem as @mvasilkov has. I click the github copilot icon, but nothing happens (no window with options is displayed) and copilot continues working. This isn't the first time that this has happened to me, but `vscode` gets updated frequently, and a few days ago this option did work as expected, altough I coudn't say which version I was using.

**Reply by davecheney** [reply]
@matrs can I ask you to double check that you are using VSCode. Linux distributions often bundle OSS Code or VSCodium and it can be sometimes not obvious that they have done so.

**Reply by matrs** [reply]
@davecheney  Im using  the package [`visual-studio-code-bin 1.71.1-1`](https://aur.archlinux.org/packages/visual-studio-code-bin), which in the official wiki is listed as 
[- **Visual Studio Code** — Proprietary Microsoft-branded release. ](https://wiki.archlinux.org/title/Visual_Studio_Code#Installation)

The [`PKGBUILD` ](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=visual-studio-code-bin) from the first link shows:
` source_x86_64=(code_x64_${pkgver}.tar.gz::https://update.code.visualstudio.com/${pkgver}/linux-x64/stable)`

Also, when I disable all the extensions except `copilot`, this problem still exists.

**Reply by davecheney** [reply]
Thanks for confirming. I’m sorry I had to ask, this does come up quite often

**Reply by davecheney** [reply]
@matrs i'm sorry, I don't have any other suggestions. As far as I'm aware the enabled/disable behaviour is one of the oldest features in the product and certainly isn't under active development so I wouldn't expect it to work or not work based on the point release you're using.

**Reply by mvasilkov** [reply]
I'm also on VS Code as downloaded from a Microsoft website. Intel Mac.

I think I figured it out, there seems to be an interaction between the Do Not Disturb mode and Copilot's prompts, where even after disabling the DND the prompts still cannot appear. After clicking on the DND mode toggle several times it seems to have started working correctly.

**Reply by matrs** [reply]
I checked my "do not disturb mode" and It was activated. Probably I did it by accident, because I didn't even knew it existed. Now, I see the copilot prompt.

**Reply by juliocormor** [reply]
Issue is still happening, disabling "do not disturb" showed the option again.

**Reply by jmpp** [reply]
I can confirm that disabling the "Do Not Disturb" mode works.

Like @matrs , I didn't even know that this option was existing! Thank you!

**Comment by meneguzzi** [comment]
I wonder if there is a way to disable co-pilot in GitHub itself. I don't want this kind of trash.
